### ISSUE TITLE
Fix S2699 FP: Support derivations of ExpectedExceptionBaseAttribute for MsTest

### ISSUE THREAD

**Issue by Smenus** [SonarSource/sonar-dotnet#6497]
### Description

When using a custom implementation of the MSTest `ExpectedExceptionAttribute`, S2699 doesn't recognise the attribute, resulting in a false positive.
PR inbound.

### Repro steps

```cs
namespace S2699Repro
{
    using System;
    using Microsoft.VisualStudio.TestTools.UnitTesting;

    [TestClass]
    public class UnitTest1
    {
        [TestMethod]
        [DerivedExpectedException]
        public void TestMethod1() // S2699: Add at least one assertion to this test case.
        {
            throw new Exception();
        }
    }

    public class DerivedExpectedExceptionAttribute
        : ExpectedExceptionBaseAttribute
    {
        protected override void Verify(Exception exception) { }
    }
}
```

### Expected behavior

All good, no code smells.

### Actual behavior

S2699 is triggered.
